dask/array/slicing.py

```diff
@@ -1674,7 +1674,7 @@ def block_index_shape_from_1d_bool_index(dim, loc0, loc1):
 
     @functools.lru_cache
     def n_preceeding_from_1d_bool_index(dim, loc0):
```

Comment:
The function name needs to be changed as well. Is it OK to change the name ad add a function with teh old name with a `FutureWarning` for deprecation?